### PR TITLE
Add MIPS variant to Machine type

### LINKED ISSUE #94: Can we add MIPS variant to Machine enum?
I think this is an information that dependent crates might need to know.

https://github.com/gimli-rs/object/blob/655dc57a26c8ebc3ca97b361e39a61b328c444d4/src/lib.rs#L87-L101

### PR DESCRIPTION
**Pull request by lzutao**
Closes #94 .

### CODE REVIEW

File: src/elf.rs [hunk 1]
```diff
@@ -157,6 +157,9 @@ where
             elf::header::EM_AARCH64 => Machine::Arm64,
             elf::header::EM_386 => Machine::X86,
             elf::header::EM_X86_64 => Machine::X86_64,
+            elf::header::EM_MIPS | elf::header::EM_MIPS_RS3_LE | elf::header::EM_MIPS_X => {
+                Machine::X86_64
```
Comment (bjorn3) [inline src/elf.rs]:
```suggestion
                Machine::Mips
```

Comment (lzutao) [reply src/elf.rs]: That's lame! :grin:  Thanks.

File: src/elf.rs [hunk 2]
```diff
@@ -157,6 +157,9 @@ where
             elf::header::EM_AARCH64 => Machine::Arm64,
             elf::header::EM_386 => Machine::X86,
             elf::header::EM_X86_64 => Machine::X86_64,
+            elf::header::EM_MIPS | elf::header::EM_MIPS_RS3_LE | elf::header::EM_MIPS_X => {
```
Comment (philipc) [inline src/elf.rs]: MIPS-X is not the same as MIPS.

Comment (philipc) [reply src/elf.rs]: And on second thoughts, please delete the `EM_MIPS_RS3_LE` too unless you have an immediate need for it (which is unlikely).

**Pull request merged**